test/quelpa-test.el

```diff
@@ -83,6 +83,24 @@
     (setq hash (cdr stamp-info))
     (should (equal (quelpa-check-file-hash "foobar") stamp-info))))
 
+(quelpa-deftest quelpa-check-dir-hash-test ()
+  "Make sure that old file hash is correctly compared with the new one
+  and only when it has changed the new time-stamp is returned."
+  (cl-letf* ((stamp-info '("20140413.90742" . "7e4c099e65d254f62e64b581c42ddeb3c487064b"))
+             (hash "4935a306e358cbd0d9bd200e13ceb1e44942b323")
+             ((symbol-function 'package-build--read-from-file) (lambda (file) stamp-info))
```

Comment:
Just `'ignore`.  The logic behind this is simple, if the function behind the symbol is the symbol of another function, replacing it with the symbol of the `ignore' function (or an actual function) should work.